lib/Board.rb

```diff
@@ -0,0 +1,62 @@
+class Board
+    attr_accessor :board 
+    
+    def initialize()
+        @board = Array.new(3){Array.new(3)}
+    end 
+
+    def add_piece(position, piece)
+        if is_position_available(position)
+            @board[position[0]][position[1]] = piece
+            true
+        else 
+            puts "Posision is not available try another.."
+            false
+        end
+    end
+
+    def is_position_available(pos) 
+        @board[pos[0]][pos[1]].nil?
+    end
+
+    def winner?(piece)
+        diagonal_win?(piece) || horizontal_win?(piece) || veritcal_win?(piece)
+    end
+
+    def board_full?
+        @board.all? do |row|
+            row.none? {|value| value == nil }
+        end
+    end
+
+    def diagonal_win?(piece)
```

Comment:
Let us remove from board any responsibility of determining the winner of the game. 

Another thing. Is there a way to simplify how we determine if a player has already won? As it is right now we are splitting checking for winning combinations horizontally, vertically or diagonally. Since the winning combination is an array or arrays and the player's moves are also stored in arrays we can use array methods such as `all?` and `any?` to determine if a player's moves array correspond to any of the winning combinations.